### PR TITLE
Version 2.0

### PR DESCRIPTION
**Pull request by s6577t**
Nailed it.

### CODE REVIEW

**Comment by charypar** [comment]
:+1:

**Comment by jonsharratt** [comment]
:+1:

**Comment by charypar** [comment]
:shipit:?

**Comment by JoeStanton** [comment]
LGTM :ship:

**Comment by s6577t** [comment]
kthxbai

**Comment by jonsharratt** [comment]
Can we get this merged please I want to use the updated features. Thanks.

File: License [hunk 1]
```diff
@@ -1,21 +1 @@
...
-all copies or substantial portions of the Software.
-
-THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
-THE SOFTWARE.
+11111111111111111
```
Comment (s6577t) [inline License:1]: This is much more readable

File: bin/cloudapp [hunk 2]
```diff
@@ -1,62 +0,0 @@
...
-  # Say it for good measure.
-  puts "Uploaded #{x} to #{url}"
-
-  # Get the embed link.
-  url = "#{url}/#{ARGV[0].split('/').last}"
-  urls << url
-end
-
-# Copy it to your (Mac's) clipboard.
-`echo '#{urls.join(',')}' | tr -d "\n" | pbcopy`
```
Comment (s6577t) [inline bin/cloudapp:62]: much less coupling now

File: zsh/functions/h [hunk 3]
```diff
@@ -1 +0,0 @@
-cd ~/$1;
```
Comment (jonsharratt) [inline zsh/functions/h:1]: Needs more 1's.

File: bin/git-rank-contributers [hunk 4]
```diff
@@ -1,60 +0,0 @@
...
-
-lines.sort_by { |a, c| -c }.each do |a, c|
-  a = a.obfuscate if obfuscate
-  a = a.htmlize if htmlize
-  if verbose
-    puts "#{a}: #{c} lines of diff"
-  else
-    puts a
-  end
-end
```
Comment (s6577t) [inline bin/git-rank-contributers:60]: great, less code, fewer bugs!

File: bin/git-wtf [hunk 5]
```diff
@@ -1,364 +0,0 @@
...
-if $key
-  puts
-  puts KEY
-end
-
-puts if modified || uncommitted
-puts "#{red "NOTE"}: working directory contains modified files." if modified
-puts "#{red "NOTE"}: staging area contains staged but uncommitted files." if uncommitted
-
-# the end!
```
Comment (s6577t) [inline bin/git-wtf:364]: really cleaned this up :smile_cat:

File: zsh/functions/gf [hunk 6]
```diff
@@ -1,4 +0,0 @@
-gf() {
-  local branch=$1
-  git checkout -b $branch origin/$branch
```
Comment (jonsharratt) [inline zsh/functions/gf:3]: Tut tut

File: bin/movieme [hunk 7]
```diff
@@ -1,30 +0,0 @@
...
-# Creates an animated gif from a movie file. Uploads to CloudApp. You must also
-# have `gifme` and `cloudapp` in your $PATH.
-#
-#   $1 - the path to the movie we're converting.
-#   $2 - the start time of the finished product.
-#        This can be in seconds, or it also accepts the "hh:mm:ss[.xxx]" format.
-#   $3 - the duration of the video sequence.
-#        This can be in seconds, or it also accepts the "hh:mm:ss[.xxx]" format.
-#
-# Examples:
```
Comment (jonsharratt) [inline bin/movieme:14]: What is this amateur hour, who needs examples?

File: bin/git-undo [hunk 8]
```diff
@@ -1,5 +0,0 @@
-#!/bin/sh
-#
-# Undo your last commit, but don't throw away your changes
```
Comment (jonsharratt) [inline bin/git-undo:3]: You shouldn't be making mistakes, we don't need undo.

File: bin/search [hunk 9]
```diff
@@ -1,8 +0,0 @@
-#!/bin/sh
-#
-# Quick search in a directory for a string ($1).
```
Comment (jonsharratt) [inline bin/search:3]: Removed this as we should be using Google for search.

File: golang/go.zsh [hunk 10]
```diff
@@ -1,2 +0,0 @@
-export GOPATH=$(cd ~/golang; pwd)
-export PATH=$PATH:$GOPATH/bin
```
Comment (jonsharratt) [inline golang/go.zsh:2]: This file was marked as `go.zsh` so made it go... away.

File: bin/git-up [hunk 11]
```diff
@@ -1,36 +0,0 @@
-#!/bin/sh
-#
-# Usage: git-up
-#        git-reup
-#
-# Like git-pull but show a short and sexy log of changes
```
Comment (s6577t) [inline bin/git-up:6]: this kind of language is not appropriate

File: git/gitconfig.symlink [hunk 12]
```diff
@@ -1,42 +0,0 @@
...
-[push]
-        # See `git help config` (search for push.default)
-        # for more information on different options of the below setting.
-        #
-        # Setting to git 2.0 default to surpress warning message
-        default = current
-[credential]
-	helper = osxkeychain
-[github]
-	user = joestanton
```
Comment (s6577t) [inline git/gitconfig.symlink:42]: this whole file is completely pointless

File: bin/movieme [hunk 13]
```diff
@@ -1,30 +0,0 @@
...
-rm -rf /tmp/movieme
-
-# create tmp dir
-mkdir -p /tmp/movieme
-
-# split the movie into constituent frames
-ffmpeg -i $1 -f image2 -ss $2 -t $3 -r 7 /tmp/movieme/d-%05d.png
-
-# ANIMATE
-gifme /tmp/movieme/* -d 0
```
Comment (jonsharratt) [inline bin/movieme:30]: Need to spend less time making animated gif movies and more time coding, removed this a lot better.

File: zsh/functions/c [hunk 14]
```diff
@@ -1 +0,0 @@
-cd $PROJECTS/$1;
```
Comment (jonsharratt) [inline zsh/functions/c:1]: Needs more 1's

**Comment by JoeStanton** [comment]
Thanks for the feedback guys, please ensure your changes conform to the style guide used on this project (deletions are prohibited). If you get this cleaned up LGTM.

File: zsh/functions/c (hunk 14 above)
Comment (JoeStanton) [inline zsh/functions/c:1]: I think more 1's could break this. Can you write a unit test?

File: bin/git-up (hunk 11 above)
Comment (JoeStanton) [inline bin/git-up:6]: Soz

File: bin/git-wtf (hunk 5 above)
Comment (JoeStanton) [inline bin/git-wtf:364]: Definitely cleaner!

**Comment by charypar** [comment]
Well ok, can we at least get the Readme merged?

**Comment by jonsharratt** [comment]
I find this README more informative, would appreciate the merge.

**Comment by JoeStanton** [comment]
Oh are we doing this again? :trollface:

**Comment by jonsharratt** [comment]
Any update on this, @sjltaylor @charypar are looking to implement this soon on their machines

File: zsh/functions/_brew [hunk 15]
```diff
@@ -1,65 +0,0 @@
-#compdef brew
```
Comment (jonsharratt) [inline zsh/functions/_brew:1]: Milk and two sugars please

File: zsh/config.zsh [hunk 16]
```diff
@@ -1,48 +0,0 @@
...
-setopt NO_BG_NICE # don't nice background tasks
-setopt NO_HUP
-setopt NO_LIST_BEEP
-setopt LOCAL_OPTIONS # allow functions to have local options
-setopt LOCAL_TRAPS # allow functions to have local traps
-setopt HIST_VERIFY
-setopt SHARE_HISTORY # share history between sessions ???
-setopt EXTENDED_HISTORY # add timestamps to history
-setopt PROMPT_SUBST
-setopt CORRECT
```
Comment (jonsharratt) [inline zsh/config.zsh:27]: INCORRECT

**Comment by charypar** [comment]
Yea, please @JoeStanton. I'd hate having to fork the project to get this functionality...

File: vim/vimrc.after.symlink [hunk 17]
```diff
@@ -1,16 +0,0 @@
-color solarized
```
Comment (jonsharratt) [inline vim/vimrc.after.symlink:1]: Energy efficient, nice one!

File: bin/git-wtf [hunk 18]
```diff
@@ -1,364 +0,0 @@
...
-if $dump_config
-  puts $config.to_yaml
-  exit
-end
-
-## first, index registered remotes
-remotes = `git config --get-regexp ^remote\.\*\.url`.split(/[\r\n]+/).inject({}) do |hash, l|
-  l =~ /^remote\.(.+?)\.url (.+)$/ or next hash
-  hash[$1] ||= $2
-  hash
```
Comment (jonsharratt) [inline bin/git-wtf:282]: Looks illegal...

File: vim/vimrc.after.symlink (hunk 17 above)
Comment (charypar) [inline vim/vimrc.after.symlink:1]: Well actually it's a huge burden on the `env` to create the theme, which is why we suggest removing.

File: zsh/config.zsh (hunk 16 above)
Comment (charypar) [inline zsh/config.zsh:27]: Yes, CORRECT was very misleading here.

Comment (jonsharratt) [reply zsh/config.zsh:27]: Wait is this INCORRECT or CORRECT now?

File: zsh/zshrc.symlink [hunk 19]
```diff
@@ -1,53 +0,0 @@
...
-# use .localrc for SUPER SECRET CRAP that you don't
-# want in your public, versioned repo.
-if [[ -a ~/.localrc ]]
-then
-  source ~/.localrc
-fi
-
-# initialize autocomplete here, otherwise functions won't be loaded
-autoload -U compinit
-compinit
```
Comment (jonsharratt) [inline zsh/zshrc.symlink:40]: Way too gangster!

File: zsh/zshrc.symlink [hunk 20]
```diff
@@ -1,53 +0,0 @@
...
-  source $file
-done
-
-# load everything but the path and completion files
-for file in ${${config_files:#*/path.zsh}:#*/completion.zsh}
-do
-  source $file
-done
-
-# use .localrc for SUPER SECRET CRAP that you don't
```
Comment (jonsharratt) [inline zsh/zshrc.symlink:31]: Offensive language!

**Comment by JoeStanton** [comment]
Old bants is the best bants eh?

**Pull request closed without merging**